NickvisionCavalier.Shared/Models/Renderer.cs

```diff
@@ -39,21 +68,62 @@ public void Draw(float[] sample, float width, float height)
         };
         if (Configuration.Current.Mirror == Mirror.Full)
         {
-            _drawFunc(sample, Configuration.Current.Direction, 0, 0, GetMirrorWidth(width), GetMirrorHeight(height), fgPaint);
+            _drawFunc(sample, Configuration.Current.Direction, Configuration.Current.AreaMargin, Configuration.Current.AreaMargin, GetMirrorWidth(width), GetMirrorHeight(height), fgPaint);
             _drawFunc(sample, GetMirrorDirection(), GetMirrorX(width), GetMirrorY(height), GetMirrorWidth(width), GetMirrorHeight(height), fgPaint);
         }
         else if (Configuration.Current.Mirror == Mirror.SplitChannels)
         {
-            _drawFunc(sample.Take(sample.Length / 2).ToArray(), Configuration.Current.Direction, 0, 0, GetMirrorWidth(width), GetMirrorHeight(height), fgPaint);
+            _drawFunc(sample.Take(sample.Length / 2).ToArray(), Configuration.Current.Direction, Configuration.Current.AreaMargin, Configuration.Current.AreaMargin, GetMirrorWidth(width), GetMirrorHeight(height), fgPaint);
             _drawFunc(sample.Skip(sample.Length / 2).Reverse().ToArray(), GetMirrorDirection(), GetMirrorX(width), GetMirrorY(height), GetMirrorWidth(width), GetMirrorHeight(height), fgPaint);
         }
         else
         {
-            _drawFunc(sample, Configuration.Current.Direction, 0, 0, width, height, fgPaint);
+            _drawFunc(sample, Configuration.Current.Direction, Configuration.Current.AreaMargin, Configuration.Current.AreaMargin, width, height, fgPaint);
         }
         Canvas.Flush();
     }
 
+    /// <summary>
+    /// Creates gradient shader
+    /// </summary>
+    /// <param name="colorStrings">List of colors as strings</param>
+    /// <param name="width">Canvas width</param>
+    /// <param name="height">Canvas height</param>
+    /// <returns>Skia Shader</returns>
+    private SKShader CreateGradient(List<string> colorStrings, float width, float height, uint margin = 0)
+    {
+        var colors = colorStrings.Select(c => SKColor.Parse(c)).Reverse().ToArray();
+        if (Configuration.Current.Mirror != Mirror.Off)
+        {
+            var mirrorColors = new SKColor[colors.Length * 2];
+            if (Configuration.Current.Direction == DrawingDirection.BottomTop || Configuration.Current.Direction == DrawingDirection.RightLeft)
+            {
+                Array.Reverse(colors);
+            }
+            colors.CopyTo(mirrorColors, 0);
+            Array.Reverse(colors);
+            colors.CopyTo(mirrorColors, colors.Length);
+            colors = mirrorColors;
+        }
+        SKShader shader;
```

Comment:
So you can using a tradition switch statement here which requires you to make that temporary `SKShader shader` variable, then set it in the switch, then return it. We can use a newer C# feature called "switch expressions" which will let you get rid of all of this and just make it:
```csharp
return Configuration.Current.Direction switch 
{
    DrawingDirection.TopBottom => SKShader.CreateLinearGradient(new SKPoint(margin, margin), new SKPoint(margin, height), colors, SKShaderTileMode.Clamp),
    DrawingDirection.BottomTop => SKShader.CreateLinearGradient(new SKPoint(margin, height), new SKPoint(margin, margin), colors, SKShaderTileMode.Clamp),
    DrawingDirection.LeftRight => SKShader.CreateLinearGradient(new SKPoint(margin, margin), new SKPoint(width, margin), colors, SKShaderTileMode.Clamp),
    _ => SKShader.CreateLinearGradient(new SKPoint(width, margin), new SKPoint(margin, margin), colors, SKShaderTileMode.Clamp)
};
Much cleaner IMO. Notice how the statements end with `,` instead of `;` and the last statement ends in nothing. Also notice how `default` just becomes `_`
```